### PR TITLE
CI: Update Rust SDK conformance test cache version

### PR DESCRIPTION
**Pull request by aLekSer**
Update cache for Cloud Builds.
As it could be seen from #1125 cloud runs, Rust sdk test downloading dependencies, while it should use cache for that in order to decrease overall CI build.
Fix documentation.

### CODE REVIEW

**Comment by markmandel** [comment]
What do you mean by "bringing it back" ? Do you mean clearing it somehow?

**Comment by aLekSer** [comment]
I would update this PR after first test run to make sure that second one is utilising the cache of first run.

**Comment by aLekSer** [comment]
@markmandel I mean that it stops working. It uses only  `test/sdk/rust/Cargo.toml` check sum.
    - '--key=$_RUST_SDK_BUILD_CACHE_KEY-$( checksum test/sdk/rust/Cargo.toml )'
So it does not update the cache if key still the same.

**Comment by aLekSer** [comment]
Triggered one more build with the same key. Cache should be used.

**Comment by aLekSer** [comment]
Comparing the results:
Rust build now spent 1 minutes less on second than on a first run.
If not used cache we had (first run):
```
+ cargo fetch
Updating crates.io index
Downloading crates ...
Downloaded cfg-if v0.1.10
Downloaded futures v0.1.29
Downloaded backtrace v0.3.40
Downloaded backtrace-sys v0.1.32
... [skipped lines]
+ cargo build
Compiling protobuf v2.0.6
Compiling cc v1.0.46
... [skipped lines]
Compiling agones v0.1.0 (/go/src/agones.dev/agones/sdks/rust)
```
Second run (does not need to perform steps above every time):
```
+ cargo fetch
Updating crates.io index
+ cargo build
Compiling agones v0.1.0 (/go/src/agones.dev/agones/sdks/rust)
```

**Comment by markmandel** [comment]
Should we rename this PR to be more indicative of what it's doing?

**Comment by markmandel** [comment]
Or should it be closed - I'm actually not sure what it's doing :smile:

**Comment by aLekSer** [comment]
This PR is needed to make Rust builds faster and Rust SDK conformance test now from #1122 don't need to download and compile third party dependencies.
In recent #1125 @roberthbailey requested this update because `cargo fetch`  saw that we have updated Agones Rust SDK and it started to download and compile all other dependencies:
```
Updating crates.io index
Downloading crates ...
Downloaded cc v1.0.46
...
Compiling agones v0.1.0 (/go/src/agones.dev/agones/sdks/rust)
```
As it could be seen in this logs:
https://console.cloud.google.com/cloud-build/builds/4b1bf464-c8df-41c8-bb03-c3eb21a1806a?project=agones-images

File: test/sdk/rust/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@
 # when Rust SDK or Rust test would be updated
 [package]
 name = "rust-simple"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (markmandel) [inline test/sdk/rust/Cargo.toml]: What is this doing? Why the version increase?

**Comment by aLekSer** [comment]
Hash of this file is used as a key in cloud build cache. After Robert changes #1125 , we got new version. However, as it could be seen from the CI logs, cache was not updated. And I was not able to find a way to make checksum of all files in Rust SDK directory, so I choose this way of forcing cache updates. Since we do not change Rust code very often.

**Pull request merged**